internal/serve/httphandler/disbursement_handler.go

```diff
@@ -103,14 +103,28 @@ func (d DisbursementHandler) PostDisbursement(w http.ResponseWriter, r *http.Req
 		return
 	}
 
-	// Get Wallet
-	wallet, err := d.Models.Wallets.Get(ctx, req.WalletID)
-	if err != nil {
-		httperror.BadRequest("wallet ID could not be retrieved", err, nil).Render(w)
-		return
+	var wallet *data.Wallet
+	if req.RegistrationContactType.IncludesWalletAddress {
+		wallets, findWalletErr := d.Models.Wallets.FindWallets(ctx, data.NewFilter(data.FilterUserManaged, true))
+		if findWalletErr != nil {
+			httperror.InternalError(ctx, "Cannot get wallets", findWalletErr, nil).Render(w)
+			return
+		}
```

Comment:
Shouldn't this filter include only enabled wallets?

Reply:
addressed here 3524434e30e7d41dd60fb0b30961ffd6ae3e0cc5